rbac/management/cache.py

```
@@ -405,6 +411,164 @@ def delete_all_principals_for_tenant(self, org_id: str):
+
+    def set_cache(self, pipe: Pipeline, key: str, item):
+        """Set workspace model to cache."""
+        pipe.set(name=key, value=pickle.dumps(item))
+        pipe.expire(name=key, time=settings.PRINCIPAL_CACHE_LIFETIME)
+        pipe.execute()
+
+    def get_from_redis(self, key: str):
+        """Get workspace from redis."""
+        obj = self.connection.get(name=key)
+        if obj:
+            return pickle.loads(obj)
```

Ruff (0.15.20)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)